### PR TITLE
Add_ServiceMonitor 

### PR DESCRIPTION
**Pull request by yangzb09**
Add ServiceMonitor to dashbase template.

### CODE REVIEW

**Comment by yangzb09** [comment]
You can find targets here:
http://prometheus.staging.zbtest.dashbase.io/targets
And grafana url here: 
admin / prom-operator
http://grafana.staging.zbtest.dashbase.io/

File: dashbase/templates/ServiceMonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+spec:
+  namespaceSelector:
+    matchNames:
+    - {{ .Release.Namespace }}
+  selector:
+    matchLabels:
+      app: {{ .Release.Name }}
+      component: table
+  endpoints:
+    - port: "admin"
```
Comment (khou) [inline dashbase/templates/ServiceMonitor.yaml]: How are these ports defined?

File: dashbase/templates/ServiceMonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+      interval: 30s
```
Comment (khou) [inline dashbase/templates/ServiceMonitor.yaml]: interval should be made into a variable in values YAML, and the default can be set to 15 or 30 seconds.

File: dashbase/templates/ServiceMonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      app: {{ .Release.Name }}
+      component: web
+  endpoints:
+    - port: "admin"
+      interval: 30s
+      honorLabels: true
+      path: /admin/prometheus-metrics
+      relabelings:
+        - sourceLabels: [__meta_kubernetes_service_label_app]
+          targetLabel: app
```
Comment (khou) [inline dashbase/templates/ServiceMonitor.yaml]: how come this doesn't need
```
- sourceLabels: [__meta_kubernetes_service_label_component]
  targetLabel: component
```

File: dashbase/templates/ServiceMonitor.yaml [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+          targetLabel: component
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: dashbase-k8s
+  namespace: {{ .Release.Namespace }}
+  labels:
+    deployment: dashbase
+    component: k8s
```
Comment (khou) [inline dashbase/templates/ServiceMonitor.yaml]: what component is this?

File: dashbase/values.yaml [hunk 5]
```diff
@@ -6,6 +6,9 @@ https: false
 
 enablePodPriority: false
 
+ServiceMonitor:
```
Comment (khou) [inline dashbase/values.yaml]: nit: let's change this to `enableServiceMonitors`

**Comment by khou** [comment]
etcd metrics are not properly scraped, but its lower priority.
Did you verify pushgateway scraping works? I didn't see collection metrics/dashboard in the deployed Grafana.

File: dashbase/templates/ServiceMonitor.yaml (hunk 1 above)
Comment (yangzb09) [reply dashbase/templates/ServiceMonitor.yaml]: Ports defined in Service, for example, web service:
```
spec:
  clusterIP: 10.0.220.95
  ports:
  - name: admin
    port: 8081
    protocol: TCP
    targetPort: 8081
  - name: http
    port: 8080
    protocol: TCP
    targetPort: 8080
  selector:
    app: zbtest
    component: web
  sessionAffinity: None
  type: ClusterIP
```
selector match service lable, and use port name as servicemonitor's endpoint.

File: dashbase/templates/ServiceMonitor.yaml (hunk 4 above)
Comment (yangzb09) [reply dashbase/templates/ServiceMonitor.yaml]: I removed it

File: dashbase/templates/ServiceMonitor.yaml (hunk 3 above)
Comment (yangzb09) [reply dashbase/templates/ServiceMonitor.yaml]: I add it in, is it same to
```

**Comment by yangzb09** [comment]
Pushgateway scraping and metircs of filebeat-monitoring dashboard has been tested.
http://grafana.staging.zbtest.dashbase.io/d/scIHJuaCiz/filebeat-monitoring?orgId=1&refresh=1m.

File: dashbase/templates/ServiceMonitor.yaml (hunk 1 above)
Comment (khou) [reply dashbase/templates/ServiceMonitor.yaml]: Thanks for clearing that up!

File: dashbase/values.yaml [hunk 6]
```diff
@@ -6,6 +6,10 @@ https: false
 
 enablePodPriority: false
 
+enableServiceMonitors: true
+
+ServiceMonitors_interval: 15s
```
Comment (khou) [inline dashbase/values.yaml]: sorry for the premature suggestion, but looks like we should change this so it's nested

e.g.
```
prometheus_service_monitors:
  enabled: true
  scrap_interval: 15s
```

**Pull request merged**